core/src/test/java/io/grpc/DeadlineTest.java

```diff
@@ -69,15 +71,19 @@ public void shortDeadlineEventuallyExpires() throws Exception {
     assertTrue(deadline.timeRemaining(TimeUnit.NANOSECONDS) > 0);
     assertFalse(deadline.isExpired());
     Thread.sleep(101);
+
     assertTrue(deadline.isExpired());
-    assertEquals(0, deadline.timeRemaining(TimeUnit.NANOSECONDS), delta);
+    long actualDelta = deadline.timeRemaining(NANOSECONDS);
+    assertTrue(maxDelta + " < " + actualDelta, maxDelta >= actualDelta);
```

Comment:
Take a second look.  It isn't calling that method.  It is calling the float version, which is not big enough to store the full 64 bit long values.  The flakiness is due to integer truncation, not time.